ui/lib/config-ui/addon/components/messages/page/create-and-edit.js

```diff
@@ -60,29 +66,33 @@ export default class MessagesList extends Component {
     try {
       this.userConfirmation = '';
 
+      const { message } = this.args;
       const isValid = this.validate();
-      const modalMessages = this.args.messages?.filter((message) => message.type === 'modal') || [];
-      const hasExpiredModalMessages = modalMessages.every((message) => {
-        if (!message.endTime) return false;
-        return isAfter(timestamp.now(), new Date(message.endTime));
-      });
 
-      if (!hasExpiredModalMessages && this.args.hasSomeActiveModals && this.args.message.type === 'modal') {
+      if (!this.hasExpiredModalMessages && this.hasSomeActiveModals && message.type === 'modal') {
         this.showMultipleModalsMessage = true;
         const isConfirmed = yield this.getUserConfirmation.perform();
         if (!isConfirmed) return;
       }
 
       if (isValid) {
-        const { isNew } = this.args.message;
-        const { id, title } = yield this.args.message.save();
-        this.flashMessages.success(`Successfully ${isNew ? 'created' : 'updated'} ${title} message.`);
-        this.pagination.clearDataset('config-ui/message');
+        const { data } = message.toJSON();
+        const method = message.isNew ? 'createCustomMessage' : 'uiConfigUpdateCustomMessage';
```

Comment:
since isNew is a conditional below, do we still need to dynamically set the message?

Reply:
Ya good call this definitely doesn't look right. Good catch! I removed the method const and left the if/else block.